includes/utilities.inc

```diff
@@ -134,13 +166,111 @@ EOQ;
     );
   };
   $issues = array_map($map_results, $results);
-  // Grab the PIDs...
-  $get_pid = function($o) {
-    return $o['pid'];
+  return $issues;
+}
+
+/**
+ * Gets all the issues that the given newspaper owns using Solr.
+ *
+ * @param AbstractObject $object
+ *   An AbstractObject representing a Fedora object.
+ *
+ * @return array
+ *   An array with elements of the form of:
+ *     - pid: The unique persistent identifier for the issue.
+ *     - label: A descriptive label for the issue.
+ *     - sequence: The sequence number of the issue, starts at 1.
+ *     - issued: A DateTime object repersenting the date the issue was released.
+ */
+function islandora_newspaper_get_issues_solr(AbstractObject $object) {
+  $parent_field = variable_get('islandora_newspaper_parent_issue_solr_field', 'RELS_EXT_isMemberOf_uri_ms');
+  $date_field = variable_get('islandora_newspaper_solr_date_field', 'RELS_EXT_dateIssued_literal_ms');
+  $sequence_field = variable_get('islandora_newspaper_solr_sequence_field', 'RELS_EXT_isSequenceNumber_literal_ms');
+  // Counter for looping the start.
+  $start = -1;
+  // Number of records to return at once.
+  $rows = 10000;
+  $results = array();
+
+  $solr_build = new IslandoraSolrQueryProcessor();
+  $solr_query = format_string('!parent_field:("info:fedora/!pid" "!pid") AND'
+    . ' RELS_EXT_hasModel_uri_ms:"info:fedora/islandora:newspaperIssueCModel"',
+    [
+      '!parent_field' => $parent_field,
+      '!pid' => $object->id,
+    ]
+  );
+
+  do {
+    $start += 1;
+    $solr_param = [
+      'rows' => $rows,
+      'fl' => format_string('!date, !sequence, PID, fgs_label_s', [
+        '!date' => $date_field,
+        '!sequence' => $sequence_field,
+      ]),
+      'start' => ($rows * $start),
+      'hl' => 'false',
+      'facet' => 'false',
+    ];
+
+    $solr_build->buildQuery($solr_query, $solr_param);
+    $solr_build->solrParams = array_replace_recursive($solr_build->solrParams, $solr_param);
+
+    try {
+      $solr_build->executeQuery();
+      $count = $solr_build->islandoraSolrResult['response']['numFound'];
+      $results = array_merge($results, $solr_build->islandoraSolrResult['response']['objects']);
+    }
+    catch (Exception $error) {
+      drupal_set_message(check_plain(t('Error searching Solr index')) . ' ' . $error->getMessage(), 'error');
+    }
+
+  } while ($count > ($rows * $start + $rows) && !isset($error));
```

Comment:
should $count be initialized to something?

Reply:
Line 222 above, basically in the case that you have more than 10,000 issues I wanted to start looping to grab them. So the count tells me the total number but only after we have done a query. I could initialize it before the loop if you think that is better. I added the `!isset($error)` to halt the loop if you get an error from Solr.

Reply:
yeah, the !isset($error) is a good one. I was just afraid that at runtime, PHP 7.x could throw some warnings because $count is undefined. I have no strong opinion really but want to be sure the logic does not allow infinite looping (i'm scared of "while" statements always). Seems like the logic won't allow that. good

Reply:
Github is doing something weird. I added a `$error = 0;` to line 193, which you can see on the branch but in the PR it appears like maybe it didn't fully update? Not sure
https://github.com/uml-digitalinitiatives/islandora_solution_pack_newspaper/blob/7.x-ISLANDORA-2126/includes/utilities.inc#L193

Reply:
@whikloj weird. I have seen some caching weirdness in GitHub lately. Can you force a new hash via an ammend or a rebase? 😭

Reply:
@whikloj still don't see $error = 0; Should i wait for that?